sklearn/metrics/classification.py

```diff
@@ -1912,8 +1912,10 @@ def brier_score_loss(y_true, y_prob, sample_weight=None, pos_label=None):
     assert_all_finite(y_true)
     assert_all_finite(y_prob)
 
+    # currently, we only support binary classification
+    _check_binary_probabilistic_predictions(y_true, y_prob)
+
     if pos_label is None:
```

Comment:
I think these three lines can be replaced by `y_true = _check_binary_probabilistic_predictions(y_true, y_prob)`

Reply:
@jnothman Sorry but I don't understand your meaning. Which two lines? The core issue I want to solve here is that previous implementation use the function after y_true is binarized. Kindly provide more details please. Thanks.

Reply:
Right, I realise again that you're not using `_check_binary_probabilistic_predictions` because it does not support `pos_label`. But it probably should.

My issue is that `_check_binary_probabilistic_predictions` does processing and returns something. It's weird not to use that return value.

I think I would add to `_check_binary_probabilistic_predictions`:
```python
out = label_binarize...
if pos_label is not None:
    if pos_label != labels[-1]:
        if pos_label != labels[0]:
            raise ValueError...
        out = 1 - out
```